service/storage_proxy.cc

```diff
@@ -723,6 +723,55 @@ class storage_proxy::remote {
         auto cmd = make_lw_shared<query::read_command>(std::move(cmd1));
         auto src_ip = src_addr.addr;
         auto timeout = t ? *t : db::no_timeout;
+
+        // Verify whether read_command is provided in legacy or native reversed format by comparing
+        // schema version with table schema version. There are three options:
+        // 1. Versions are the same -> legacy format
+        // 2. schema version is equal to reversed table schema -> native format
+        // 3. table schema is outdated
+        static auto is_read_command_in_reversed_legacy_format = [](const query::read_command& cmd, schema_ptr table_schema) -> std::optional<bool> {
+            if (cmd.schema_version == table_schema->version()) {
+                return true;
+            }
+
+            if (cmd.schema_version == reversed(table_schema->version())) {
+                return false;
+            }
+
+            return std::nullopt;
+        };
+
+        // Detect whether a transformation from legacy reverse format into native reverse
+        // format is necessary before executing the read_command. That happens when the
+        // native_reverse_queries feature is turned off. A custer has mixed nodes.
+        bool format_reverse_required = false;
+
+        // Check if we have a reversed query
+        if (cmd->slice.is_reversed()) {
+            // Compare the schema version provided by read_command with the table schema replica holds.
+            auto table_schema = p->local_db().find_schema(cmd->cf_id);
```

Comment:
If this node missed a schema update, this can throw and kill the read, even though the read is supposed to survive unsynced schema, via the `get_schema_for_read()` call below.

Reply:
I managed to simulate such a situation by inserting in `migration_manager::merge_schema_from` an error injection that throws an exception. Then one of my nodes in a 3-node cluster missed the information that a table was created. 
Executing a SELECT query with CONSISTENCY ALL, throws an exception when calling `get_schema_for_read()`. It eventually calls `database::find_schema(<table id>)` and fails. I tested it even with the forward read queries to skip this entire `if-`statement. After digging around, I made it work when deliberately changed `if (use_raft)` into `if (false)` in https://github.com/scylladb/scylladb/blob/a99200931a494bb5dc5b4f86015271a6f78dc724/service/migration_manager.cc#L1223
to **not** use raft when syncing a pulled schema. 
With this slight change, it worked for both forward and reversed queries. It looks like we either have a bug or I created a super special corner case with my error injection.

Reply:
Not completely true. I think I created a super special corner case that not necessarily is valid but yet allowed me to test a situation where schema update does not reach the node.